src/bin/wit-bindgen.rs

```diff
@@ -150,8 +150,9 @@ fn gen_world(
         resolve.push(UnresolvedPackage::parse_file(&opts.wit)?)?
     };
     let world = resolve.select_world(pkg, opts.world.as_deref())?;
-    generator.generate(&resolve, world, files);
-    Ok(())
+    generator
+        .generate(&resolve, world, files)
+        .map_err(|e| anyhow!("{e} (re-run with --help for options summary)"))
```

Comment:
Oh to clarify I mostly mean that `generate` should return` Result<()>` and this can use `.with_context` or otherwise print `help: ...` information after printing the error. The fundamental ability to return an error is totally fine and no issues with that, this is instead commentary on how the error is presented here

Reply:
I've switched it to an `eprintln!` with a more specific error message.  Let me know if you think it still needs tweaking.